examples/common/imgui_ui/windows/light.cpp

```diff
@@ -65,9 +65,18 @@ ImVec4 HueSaturationToColor(float hueDegrees, float saturationPercent)
 
 void Light::UpdateState()
 {
+    if (mTargetLightIsOn.HasValue()) {
+        OnOff::Attributes::OnOff::Set(mEndpointId, mTargetLightIsOn.Value());
```

Comment:
Should this be calling `OnOffServer::setOnOffValue`?  There's a bunch of logic that has to run when the state changes other than setting this attribute...

As in, should we be pretending to be an On or Off command here instead of just writing the attribute?

Reply:
That makes sense. Looking now.

The coupling is a bit harsh there though: lots of ember direct calls and the CommandHandler object passed in to the methods is very complex (probably mocking is not a great idea). Will look on how to update.

Reply:
Updated both this and level control.